### ISSUE TITLE
Cutting outside registration marks 

### ISSUE THREAD

**Issue by GVCompu** [fablabnbg/inkscape-silhouette#124]
Dear, 

I tried the extension in inkscape to cut on my silhoette cameo3
Seems to work quiet well , thx for the effort :-)
But i have a remark/question :
I noticed it is not possible to cut outside de registration marks square.
Is this a programming choice or is it really not possible..

Maybe i better explain what i would want to do with that :-)

On an A4 page i want to cut out a drawing that has the width of the A4 minus 4mm (approx) 
So i can print that but the registration marks can only on 10mm from the border , so the cut is not full..
Without registration marks i can cut the drawing , but then it's difficult to get the position accurate enough without trying and retrying.

So my question is, as you programmed this yourself if it is totally impossible to change program so that one can work with registration marks to initialize the position of the cut-head , and then still cut outside de bounderies of the marks...

I hope you fully understand my question.

You did a great job programming this, my utter admiration :-)  

Kind regards

**Comment by EtherGraf** [comment]
With the current state of the extension it is not possible, I have found out this too. I am not sure if this is a device firmware limit or a software limit. 
Maybe with some research one can send other commands (or other parameters to the commands) and achieve the wanted result, but it will not be done with a little effort. 

There is another problem (or the root cause?) that the extension limits the cutting size to a bounding box not bigger than given device lmits. You can try to disable the option "Enable Software Clipping" and see if this improves something.

To sum up, i would not totally exclude the possibility to get this working, but with the current state and knowledge it is not possible.

**Comment by GVCompu** [comment]
Ok thnx for this quick answer.
Too bad this is not possible, it makes the cameo3 for my project useless.
And i find it so strange, for me it's not logical :-) 

Kind regards and keep up the good work

**Comment by miLORD1337** [comment]
Won't it be possible to print the regmarks a tad wider, so they incorporate your entire drawing?

**Comment by EtherGraf** [comment]
Or place your A4 sheet on another paper with a size A4+ and print the regmarks there? The device itself is not limited to A4, especially in length.

**Comment by GVCompu** [comment]
The drawing is indeed to big for the max width of the regmarks.

And yes you could place the A4 sheet on another paper and print er regmarks there, but if you can position that well (the one paper on the other) then you can simple try to position the paper on the cutting mat correct, it's possible but not always easy...

Thats the whole thint with registration marks, then you don't have to care to position your paper exactly on the mat, you don't solve that with putting it on another paper in fact

**Comment by EtherGraf** [comment]
Ok. you're right about placing paper on paper, this idea wasn't well tought.

But i am not aware of a maximum width of regmarks. The given A4 templates are only examples, you can move the marks to other positions. Afterwards It is important to set the actual positions in the extension. So it should be possible to move the marks to wieder postions. The left ones more left and the right ones to the right.
Surely there are some device limits, but they have not to be the ones from the examples. Have you tried it?

**Comment by GVCompu** [comment]
No the marks can not be placed nearer dan 10mm from the border.

**Comment by GVCompu** [comment]
I also tried to first send a job with marks to position the cameo3 cuthead, let it return to the original position and then send a job without unloading, hoping it would remember the positions 👍 
But that didn't work, but maybe this could be something to try via programming and keeping the position after registering the marks, but deceive the cameo that there are no marks anymore?

I hope someone understands what i want to say :-)  (not native English)
Kind regards

**Comment by EtherGraf** [comment]
I fully understand your idea, but I have to disappoint you. I see no way to achieve this. The handling of the regmarks takes place fully on the device, there is no position or calibration factor sent back to the extension. So it cannot handle it by itself.

But I dont understand your previous message. Why should it not possible to place the regmark nearer than 10 mm to the border? Just print it 8mm to the left (i.e. with no border) and move the paper physically 8mm to the right. From view of the device it is the same, but you have won 8mm on the rigth side. Am I wrong with this?

Ok, offset calculation would be tricky. Therefore restore the old version of the svg (with 10 mm border), set 10mm in the extension and plot.
I aggree, the offset calculation is not intuitive and should be simplified, but we have to use what we have.

**Comment by GVCompu** [comment]
Well that's because in de dialog of the extension i cannot set the position of the regmark lower than 10mm..
See attached image :-)


![image](https://user-images.githubusercontent.com/77343089/104510641-beb2da80-55eb-11eb-8a8f-083ba733e8cc.png)

**Comment by GVCompu** [comment]
And moving het paper on the mat 8mm to the right? i don't understand that sorry

**Comment by GVCompu** [comment]
I temporaraly solved my problem by putting a few crosses in the outskirts of the A4 page, i then test with pen-tool how good the paper is positioned and adapt the offset on X and Y .. from the moment the pen is on the cross exactly i cut the rest of the page.  It is time consuming but the only way to get a predictable accurate cut-result.   

I am making a prototype machine overlay , so it has to be perfect cut out.

And again i have a small remark :-)

It would be usefull that one can save the different settings per cutting tool.
Now you have to reset them eacht time you choose another tool...
See picture.. when i select the pen tool , i have to adapt the settings to not have a too hard press with the pen .. when i want to cut and change again to autoblade, the settings are not reset to the settings wich where already set 


Anyway, it's just a remark,  I thnk you very much for the effort of answering and programming this extension.. it helped me a lot 

Kind regards :-)



![image](https://user-images.githubusercontent.com/77343089/104715791-494c2480-5727-11eb-8042-f6d7f62b35f0.png)

**Comment by ernierasta** [comment]
Hi @GVCompu it should be possible, to store different settings for different tools. Trick is, to make Inkscape think you have as many extensions as you have your setting profiles (tools). Trick is described for countersheet extension:
https://github.com/lifelike/countersheetsextension/wiki/Multiple-Configurations
But it will work also with this extension.
